include/nbl/builtin/hlsl/sampling/quotient_and_pdf.hlsl

```diff
@@ -0,0 +1,50 @@
+// Copyright (C) 2018-2025 - DevSH Graphics Programming Sp. z O.O.
+// This file is part of the "Nabla Engine".
+// For conditions of distribution and use, see copyright notice in nabla.h
+
+#ifndef _NBL_BUILTIN_HLSL_SAMPLING_QUOTIENT_AND_PDF_INCLUDED_
+#define _NBL_BUILTIN_HLSL_SAMPLING_QUOTIENT_AND_PDF_INCLUDED_
+
+#include "nbl/builtin/hlsl/concepts/vector.hlsl"
+#include "nbl/builtin/hlsl/vector_utils/vector_traits.hlsl"
+
+namespace nbl
+{
+namespace hlsl
+{
+namespace sampling
+{
+
+template<typename T>
+NBL_BOOL_CONCEPT Spectral = (concepts::Vectorial<T> || concepts::Scalar<T>) && is_floating_point_v<typename vector_traits<T>::scalar_type>;
+
+// finally fixed the semantic F-up, value/pdf = quotient not remainder
+template<typename Q, typename P NBL_PRIMARY_REQUIRES(Spectral<Q> && is_floating_point_v<P>)
+struct quotient_and_pdf
+{
+    using this_t = quotient_and_pdf<Q, P>;
+    static this_t create(NBL_CONST_REF_ARG(Q) _quotient, NBL_CONST_REF_ARG(P) _pdf)
+    {
+        this_t retval;
+        retval.quotient = _quotient;
+        retval.pdf = _pdf;
+        return retval;
+    }
+
+    Q value()
+    {
+        return quotient*pdf;
+    }
+
+    Q quotient;
+    P pdf;
+};
+
+typedef quotient_and_pdf<float32_t, float32_t> quotient_and_pdf_scalar;
+typedef quotient_and_pdf<vector<float32_t, 3>, float32_t> quotient_and_pdf_rgb;
```

Comment:
either make them template aliases on `T` and use that instead of `float32_t` or add `32` suffix to the aliases

Reply:
also these typedefs belong in the `bxdf` namespace https://github.com/Devsh-Graphics-Programming/Nabla/pull/811#discussion_r2048109312